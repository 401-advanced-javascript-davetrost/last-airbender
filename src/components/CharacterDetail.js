import React from 'react';
import PropTypes from 'prop-types';
import { getCharacters } from '../services/getCharacters';

export default class CharacterDetail extends React.Component {

  state = { 
    name: '', 
    photoUrl: '', 
    gender: '',
    hair: '',
    position: '',
    allies: [], 
    enemies: [], 
    affiliation: '', 
  }

  handleClose = () => {
    const { history } = this.props;
    history.goBack();
  }

  componentDidMount = () => {
    const { id } = this.props.match.params;
    getCharacters({ id })
      .then(item => this.setState({ ...item }));
  }

  render() {
    const { photoUrl, gender, hair, position, allies, enemies, affiliation } = this.state;
    return (
      <div>
        <h1>{name}</h1>
        <img src={photoUrl} alt={'image of ' + name} />
        <dl>
          {gender && <dd>Gender: {gender}</dd>}
          {hair && <dd>Hair: {hair}</dd>}
          {position && <dd>Position: {position}</dd>}
          {allies && <dd>Allies: {allies.join(', ')}</dd>}
          {enemies && <dd>Enemies: {enemies.join(', ')}</dd>}
          <dd>Affiliation: {affiliation || 'unknown'}</dd>
        </dl>
        <button onClick={this.handleClose}>Close</button>
      </div>
    );
  }

  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string.isRequired,
      }).isRequired
    }).isRequired,
    history: PropTypes.object.isRequired,
  }

}
