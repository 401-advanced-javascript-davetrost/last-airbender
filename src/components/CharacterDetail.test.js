import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail';

jest.mock('../services/getCharacters', () => ({
  getCharacters() {
    return Promise.resolve([
      {
        _id: '5cf5679a915ecad153ab6927',
        allies: ['Her husband'],
        enemies: ['Amon'],
        photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/f/f9/Fire_Nation_councilwoman.png/revision/latest?cb=20121110144847',
        name: 'Fire Nation councilwoman',
        gender: 'Female',
        hair: 'Gray',
        love: 'Her husband',
        weapon: 'Fire',
        profession: 'Councilwoman ',
        position: 'Fire Nation representative to the United Republic Council (formerly) Upper-class United Republic citizen',
        affiliation: 'Fire Nation United Republic Council (formerly)',
      }
    ]);
  }
}));


describe('Character Detail component', () => {
  it('renders character details', () => {
    const wrapper = shallow(<CharacterDetail match={{ params: { id: '12345' } }} history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
