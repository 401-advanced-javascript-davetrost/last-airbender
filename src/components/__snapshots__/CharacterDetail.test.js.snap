// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail component renders character details 1`] = `
<div>
  <h1>
    nodejs
  </h1>
  <img
    alt="image of nodejs"
    src=""
  />
  <dl>
    <dd>
      Allies: 
    </dd>
    <dd>
      Enemies: 
    </dd>
    <dd>
      Affiliation: 
      unknown
    </dd>
  </dl>
  <button
    onClick={[Function]}
  >
    Close
  </button>
</div>
`;
